egs/wsj/s5/steps/make_phone_graph.sh

```diff
@@ -92,17 +92,14 @@ fi
 
 if [ $stage -le 3 ]; then
   echo "$0: creating G_phones.fst from ARPA"
-  gunzip -c $dir/phone_graph/arpa_noug.gz | arpa2fst - - | fstprint | \
-    utils/eps2disambig.pl | utils/s2eps.pl | \
-    awk '{if (NF < 5 || $5 < 100.0) { print; }}' | \
-    fstcompile --isymbols=$lang/phones.txt --osymbols=$lang/phones.txt \
-       --keep_isymbols=false --keep_osymbols=false | \
-    fstconnect | \
-    fstrmepsilon > $dir/phone_graph/G_phones.fst
-   fstisstochastic $dir/phone_graph/G_phones.fst  || echo "[info]: G_phones not stochastic."
+  gunzip -c $dir/phone_graph/arpa_noug.gz | \
+    arpa2fst --disambig-symbol=#0 --read-symbol-table=$lang/phones.txt - - | \
+    fstprint | awk '{if (NF < 5 || $5 < 100.0) { print; }}' | fstcompile | \
```

Comment:
On Mon, Apr 11, 2016 at 2:19 PM, Kirill Katsnelson <notifications@github.com
> wrote:

> In egs/wsj/s5/steps/make_phone_graph.sh
> <https://github.com/kaldi-asr/kaldi/pull/682#discussion_r59283535>:
>
> > @@ -92,17 +92,14 @@ fi
> >
> >  if [ $stage -le 3 ]; then
> >    echo "$0: creating G_phones.fst from ARPA"
> > -  gunzip -c $dir/phone_graph/arpa_noug.gz | arpa2fst - - | fstprint | \
> > -    utils/eps2disambig.pl | utils/s2eps.pl | \
> > -    awk '{if (NF < 5 || $5 < 100.0) { print; }}' | \
> > -    fstcompile --isymbols=$lang/phones.txt --osymbols=$lang/phones.txt \
> > -       --keep_isymbols=false --keep_osymbols=false | \
> > -    fstconnect | \
> > -    fstrmepsilon > $dir/phone_graph/G_phones.fst
> > -   fstisstochastic $dir/phone_graph/G_phones.fst  || echo "[info]: G_phones not stochastic."
> > +  gunzip -c $dir/phone_graph/arpa_noug.gz | \
> > +    arpa2fst --disambig-symbol=#0 --read-symbol-table=$lang/phones.txt - - | \
> > +    fstprint | awk '{if (NF < 5 || $5 < 100.0) { print; }}' | fstcompile | \
>
> How long a usage line is acceptable?
>
> Would this statement be more clear:
>
> Store symbol table with FST. Symbols are always stored if symbol tables
> are neither read or written otherwise
>
>
I wasn't talking about the options-registering string, but about the usage
message (after "usage:") for the tool as a whole.  And example command
lines (after the usage, after "e.g.:"-- you could have 'new' and 'old'
example command lines, and explain that in the 'new' one, the symbol tables
are not output, and explain the differences in the output.  Since the
behavior of this tool is much more complex than most Kaldi tools, it
deserves more explanation in the usage message.  E.g. would it be clear to
you yourself in 10 years what the program was doing?


> This complication is intended to keep arpa2fst backward-compatible. The
> old behavior of just arpa2fst input | fstprint is to print the FST with
> symbols. I could have set the default to true always, but that would be a
> wholesale inconvenience in the new invocation pattern that would have to
> carry --keep_symbols=false over every call. Hence I took the tradeoff of a
> more complex default. Arguably, this makes some intuitive sense: if an ARPA
> file is read, but no words.txt is provided and the symbols are not saved to
> a separate file, force save the symbols to FST, otherwise they will be lost
> and the FST will be useless. In another program I would just print an error
> message if an invalid combination of switches is provided, but in this case
> it does not cut it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/682/files/6e8dd7a3e849f08d04b4fb28cd3fbd772297d4bd#r59283535>
>